recipes/apriltag/all/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.1)
+project(PackageTest CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(example example.cpp)
+target_link_libraries(example ${CONAN_LIBS})
+target_link_libraries(example pthread)
```

Comment:
```suggestion
```
Conan should be able to solve it.

Reply:
If I remove this line I get: 
```
workerpool.c:(.text+0x2b2): undefined reference to `pthread_join'
collect2: error: ld returned 1 exit status
CMakeFiles/example.dir/build.make:83: recipe for target 'bin/example' failed
```
When compiling with -o apriltag:shared=False. It works when I compile the shared version of the library.

Reply:
Don't mind my previous answer, I just noticed that
`

if self.settings.os == "Linux":
--
  | self.cpp_info.system_libs = ["m", "pthread"]

`
Solves this issue.

Reply:
Don't mind my previous answer, I just noticed that
`

if self.settings.os == "Linux":
--
  | self.cpp_info.system_libs = ["m", "pthread"]

`
Fixes this.